pkg/bgpv1/manager/reconcilerv2/service.go

```diff
@@ -358,24 +271,49 @@ func (r *ServiceReconciler) getDiffRoutePolicies(p ReconcileParams, toUpdate []*
 }
 
 func (r *ServiceReconciler) getDesiredSvcRoutePolicies(p ReconcileParams, svc *slim_corev1.Service, ls sets.Set[resource.Key]) (RoutePolicyMap, error) {
-	// get cluster IP route policy
-	desiredClusterRoutePolicies, err := r.getClusterIPRoutePolicies(p, svc, ls)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get desired cluster IP route policies: %w", err)
-	}
+	desiredSvcRoutePolicies := make(RoutePolicyMap)
 
-	desiredExternalIPRoutePolicies, err := r.getExternalIPRoutePolicies(p, svc, ls)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get desired external IP route policies: %w", err)
-	}
+	for peer, afAdverts := range r.getMetadata(p.BGPInstance).ServiceAdvertisements {
+		for fam, adverts := range afAdverts {
+			agentFamily := types.ToAgentFamily(fam)
 
-	// merge two route policies, both will have unique names since name contains advertisement type
-	desiredSvcRPs := desiredClusterRoutePolicies
-	for k, v := range desiredExternalIPRoutePolicies {
-		desiredSvcRPs[k] = v
+			for _, advert := range adverts {
+				labelSelector, err := slim_metav1.LabelSelectorAsSelector(advert.Selector)
+				if err != nil {
+					return nil, fmt.Errorf("failed constructing LabelSelector: %w", err)
+				}
+				if !labelSelector.Matches(serviceLabelSet(svc)) {
+					continue
+				}
+				// LoadBalancerIP
+				policy, err := r.getLoadBalancerIPRoutePolicy(p, peer, agentFamily, svc, advert, ls)
```

Comment:
The single variable definition can cause some confusion and pointer reuse issues. Maybe use explicit policy definition names: lbPolicy, extPolicy, clusterPolicy.

Reply:
ack, done